### PR TITLE
Remove spot market + optimisations

### PR DESCRIPTION
**Pull request by 0xjocke**
- This PR removes the spot market and lets us use native ERC-20 as collateral
- Using constructor + immutable pattern for setting synthetix-core + oracle manager address
- Deployment related modifications (https://github.com/Synthetixio/synthetix-v3/pull/2153)
- Optimisations (https://github.com/Synthetixio/synthetix-v3/pull/2152)

### CODE REVIEW

File: markets/bfp-market/contracts/interfaces/IMarginModule.sol [hunk 1]
```diff
@@ -80,39 +82,41 @@ interface IMarginModule is IBasePerpMarket {
...
 
-    /// @notice Modifies the `synthMarketId` margin for `accountId` on `marketId` by the `amountDelta`. A negative amount
+    /// @notice Modifies the `collateralAddress` margin for `accountId` on `marketId` by the `amountDelta`. A negative amount
     ///         signifies a withdraw and positive is deposit. A variety of errors are thrown if limits or collateral
     ///         issues are found.
     /// @param accountId Account to modify margin collateral against
     /// @param marketId Market to modify margin collateral against
-    /// @param synthMarketId Id of modified synth asset
+    /// @param collateralAddress Address of collateral to deposit or withdraw
     /// @param amountDelta Amount of synths to deposit or withdraw
```
Comment (davidvuong) [inline markets/bfp-market/contracts/interfaces/IMarginModule.sol:91]:
```suggestion
    /// @param amountDelta Amount of collateral to deposit or withdraw
```

**Review by davidvuong — CHANGES_REQUESTED** [review]
Left some comments

File: markets/bfp-market/contracts/modules/LiquidationModule.sol [hunk 2]
```diff
@@ -80,7 +94,12 @@ contract LiquidationModule is ILiquidationModule {
 
         emit MarketSizeUpdated(marketId, updatedMarketSize, updatedMarketSkew);
 
-        (uint256 utilizationRate, ) = market.recomputeUtilization(oraclePrice);
+        (uint256 utilizationRate, ) = market.recomputeUtilization(
+            oraclePrice,
+            SYNTHETIX_CORE,
```
[the remark this answers is not in the document]
Comment (0xjocke) [reply markets/bfp-market/contracts/modules/LiquidationModule.sol:110]: Yea I was thinking of using some kinda struct. I like this!

**Review by davidvuong — APPROVED** [review]
sned it

**Pull request merged**